hyperopt/fmin.py

```diff
@@ -506,6 +520,21 @@ def fmin(
     validate_timeout(timeout)
     validate_loss_threshold(loss_threshold)
 
+    if space == "auto":
+        # Read space from objective fn
+        space = inspect.getfullargspec(fn).annotations
+
+        # Validate space
+        for var, hp_func in space.items():
+            if not isinstance(hp_func, pyll.base.Apply):
+                raise exceptions.InvalidAnnotationParameter(
+                    f"Parameter {var} has type {hp_func} which is not "
```

Comment:
given it's a parameter, what if we call it `param` ?

Reply:
Either if we make "auto" the default value for `space` or not, I would give more info in this msg error, e.g:

```
When using `space="auto"`, please annotate the objective function arguments with a `pyll.base.Apply` subclass. See example in `fmin` docstring
```

Reply:
Great suggestion